code/client/munkilib/updatecheck.py

```diff
@@ -2608,6 +2617,39 @@ def getmanifest(partialurl, suppress_errors=False):
         MANIFESTS[manifestname] = manifestpath
         return manifestpath
 
+def cleanUpManifests(subdir=""):
+    """Removes any manifest files that are no longer in use by this client"""
+    manifest_dir = os.path.join(munkicommon.pref('ManagedInstallDir'),
+                               'manifests')
+
+    exceptions = [
+        "SelfServeManifest"
+    ]
+
+    # For recursive calls (checking subfolders), append subdir to root manifest directory
+    if subdir:
+        manifest_dir = os.path.join(manifest_dir, subdir)
+
+    for item in os.listdir(manifest_dir):
+        cachename = os.path.join(subdir, item)
+
+        if cachename in exceptions:
+            continue
+            
+        if cachename not in MANIFESTS.keys():
+            path = os.path.join(manifest_dir, item)
+            if os.path.isdir(path):
+                cleanUpManifests(cachename)
+            else:
+                os.unlink(path)
+
+    # Remove folder if empty
+    try:
+        os.rmdir(manifest_dir)
```

Comment:
Might I suggest https://github.com/munki/munki/blob/master/code/client/munkilib/updatecheck.py#L2825-L2842
as a model for a slightly different implementation?